src/loaders.py

```diff
@@ -1,5 +1,162 @@
-from .steppy.pytorch.loaders import ImageSegmentationBaseDataset
+import os
+import torch
+from attrdict import AttrDict
+from sklearn.externals import joblib
+from torch.utils.data import Dataset, DataLoader
+from torch.utils.data.sampler import Sampler
+from PIL import Image
+from steppy.base import BaseTransformer
 
+from .retinanet import DataEncoder
 
-class DetectionLoader(ImageSegmentationBaseDataset):
-    pass
+
+class RandomSubsetSampler(Sampler):
+    def __init__(self, data_size, subset_size):
+        super().__init__()
+        self.data_size = data_size
+        self.subset_size = subset_size
+
+    def __iter__(self):
+        return iter(torch.randperm(self.data_size)[:self.subset_size].long())
+
+    def __len__(self):
+        return self.subset_size
+
+
+class ImageDetectionDataset(Dataset):
+    def __init__(self, X, y, labels, images_dir, encoder, train_mode):
+        super().__init__()
+        self.X = X
+        if y is not None:
+            self.y = y
+            self.labels = labels
```

Comment:
@taraspiotr I don't see any place, where you use `self.labels`. Why do you need it and what it should contain? I don't see a place in pipelines.py, where you pass `labels` to loader Step. Are you aware of that and you plan to add it in future?

Reply:
@apyskir Yes, I was aware of that, at this moment we didn't decide how are we going to use informations from labels in train time, but I wanted to have them ready, since they were already included in `data` in train()

Reply:
@taraspiotr Ok, great. Thx for the info.

Reply:
@taraspiotr I agree with you both. On one hand if something is not used it should be dropped. On the other hand we are talking about using it pretty soon. I lean on the side of having it ready